src/lib/Preview.js

```diff
@@ -782,6 +848,23 @@ class Preview extends EventEmitter {
     }
 
     /**
+     * Enrich file entries with offline links.
+     *
+     * @private
+     * @return {void}
+     */
+    enrichOfflineLinks() {
```

Comment:
We probably need to rethink this implementation since it assumes we want to replace all of the representation's URLs with one offline link. This may work for a simple document use case but breaks with multi-asset viewers like video and 3D.

Handling offline for HD video is particularly difficult since the main representation for DASH is actually a manifest which the open source library (Shaka Player) knows how to read. Shaka will then make requests to the URLs described in the manifest.

I would also move this logic to https://github.com/box/box-content-preview/blob/af861cfc33c6c4648a2cd31598c22a9dcf06819e/src/lib/file.js since we already have some logic there to manipulate representations in a file. 

Speaking of... looking at the logic there for `addOriginalRepresentation` gives me the idea that we could potentially solve this by creating `addOfflineRepresentation`, which takes in a file + an offline URL and adds an `OFFLINE` representation to the file, which is the URL you configure in `previewOptions.offlineConfig`. Then, in https://github.com/box/box-content-preview/blob/af861cfc33c6c4648a2cd31598c22a9dcf06819e/src/lib/viewers/doc/DocLoader.js, you could add two new configurations for Presentation and Document that accept this new `OFFLINE` representation.

Not sure yet - any other thoughts on how we could scope this change to only documents or single-rep viewers @JustinHoldstock @jeremypress @pramodsum @MinhHNguyen ?